packages/esbuild/test/index.js

```diff
@@ -16,177 +16,189 @@ import React from 'react'
 import {renderToStaticMarkup} from 'react-dom/server'
 import esbuildMdx from '../index.js'
 
-test('@mdx-js/esbuild', async () => {
-  // MDX.
-  await fs.writeFile(
-    new URL('esbuild.mdx', import.meta.url),
-    'export const Message = () => <>World!</>\n\n# Hello, <Message />'
-  )
-
-  await esbuild.build({
-    bundle: true,
-    define: {'process.env.NODE_ENV': '"development"'},
-    entryPoints: [fileURLToPath(new URL('esbuild.mdx', import.meta.url))],
-    outfile: fileURLToPath(new URL('esbuild.js', import.meta.url)),
-    format: 'esm',
-    plugins: [esbuildMdx()]
+test('@mdx-js/esbuild', async (t) => {
+  await t.test('MDX', async () => {
+    await fs.writeFile(
+      new URL('esbuild.mdx', import.meta.url),
+      'export const Message = () => <>World!</>\n\n# Hello, <Message />'
+    )
+
+    await esbuild.build({
+      bundle: true,
+      define: {'process.env.NODE_ENV': '"development"'},
+      entryPoints: [fileURLToPath(new URL('esbuild.mdx', import.meta.url))],
+      outfile: fileURLToPath(new URL('esbuild.js', import.meta.url)),
+      format: 'esm',
+      plugins: [esbuildMdx()]
+    })
+
+    /** @type {MDXContent} */
+    const Content =
+      /* @ts-expect-error file is dynamically generated */
+      (await import('./esbuild.js')).default // type-coverage:ignore-line
+
+    assert.equal(
+      renderToStaticMarkup(React.createElement(Content)),
+      '<h1>Hello, World!</h1>',
+      'should compile'
```

Comment:
a) I thought you didn’t want assertion messages?
b) to not have messages there, I have been moving those to `x` in `test(x, ...)`. So ``'should compile `.md`'`` instead of `'MD'`

Reply:
The esbuild tests were already somewhat categorized using comments. I used those comments as the test names. I changed it to use the previous assertion messages now, as that’s consistent with the other tests indeed.